pkg/ottl/expression.go

```diff
@@ -581,3 +582,33 @@ func (p *Parser[K]) newGetterFromConverter(c converter) (Getter[K], error) {
 		keys: c.Keys,
 	}, nil
 }
+
+// DurationGetter is a Getter that must return an time.Duration.
+type DurationGetter[K any] interface {
+	// Get retrieves an int64 value.
+	Get(ctx context.Context, tCtx K) (time.Duration, error)
+}
+
+// StandardDurationGetter is a basic implementation of DurationGetter
+type StandardDurationGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (interface{}, error)
+}
+
+// Get retrieves an int64 value.
+// If the value is not an int64 a new TypeError is returned.
+// If there is an error getting the value it will be returned.
+func (g StandardDurationGetter[K]) Get(ctx context.Context, tCtx K) (time.Duration, error) {
```

Comment:
Please add some tests in `expression_test.go` for this function.  There are similar tests for `StringGetter` that can be used as an example.